File: android/app/src/main/java/com/budgetbee/models/Transaction.java
package com.budgetbee.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.google.gson.annotations.SerializedName;

@Entity(tableName = "transactions")
public class Transaction {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @SerializedName("id")
    public int transactionId;

    @SerializedName("user_id")
    public int userId;

    @SerializedName("type")
    public String type; // "income" or "expense"

    @SerializedName("amount")
    public double amount;

    @SerializedName("category")
    public String category;

    @SerializedName("description")
    public String description;

    @SerializedName("date")
    public String date;

    @SerializedName("created_at")
    public String createdAt;

    @SerializedName("updated_at")
    public String updatedAt;

    public Transaction() {}

    public Transaction(String type, double amount, String category, String description, String date) {
        this.type = type;
        this.amount = amount;
        this.category = category;
        this.description = description;
        this.date = date;
    }

    @Override
    public String toString() {
        return "Transaction{" +
                "id=" + transactionId +
                ", type='" + type + '\'' +
                ", amount=" + amount +
                ", category='" + category + '\'' +
                ", description='" + description + '\'' +
                ", date='" + date + '\'' +
                '}';
    }
}

File: app/src/main/java/com/example/expensemanager/data/local/Transaction.kt
package com.example.expensemanager.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val username: String,
    val title: String,
    val amount: Double,
    val type: String, // "income" hoặc "expense"
    val category: String,
    val date: Long,
    val imagePath: String? = null // Thêm trường này để lưu đường dẫn ảnh hóa đơn
)
